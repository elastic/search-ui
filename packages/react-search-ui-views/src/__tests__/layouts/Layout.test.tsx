import React from "react";
import { Layout } from "../..";
import { shallow } from "enzyme";

it("renders correctly", () => {
  const wrapper = shallow(
    <Layout
      header={<div>Header</div>}
      sideContent={
        <div>
          <div>Side Content</div>
        </div>
      }
      bodyContent={<div>Body Content</div>}
      bodyHeader={<div>Body Header</div>}
      bodyFooter={<div>Body Footer</div>}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it("will accept children instead of bodyContent", () => {
  const wrapper = shallow(
    <Layout
      header={<div>Header</div>}
      sideContent={
        <div>
          <div>Side Content</div>
        </div>
      }
      bodyHeader={<div>Body Header</div>}
      bodyFooter={<div>Body Footer</div>}
    >
      <div>Body Content</div>
    </Layout>
  );
  expect(wrapper).toMatchSnapshot();
});

it("renders with className prop applied", () => {
  const wrapper = shallow(<Layout className="test-class" />);

  expect(wrapper.hasClass("test-class")).toBeTruthy();
});
