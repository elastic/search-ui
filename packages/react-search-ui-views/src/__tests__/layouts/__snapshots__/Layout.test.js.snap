// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sui-layout"
>
  <div
    className="sui-layout-header"
  >
    <div
      className="sui-layout-header__inner"
    >
      <div>
        Header
      </div>
    </div>
  </div>
  <div
    className="sui-layout-body"
  >
    <div
      className="sui-layout-body__inner"
    >
      <LayoutSidebar
        className="sui-layout-sidebar"
      >
        <div>
          <div>
            Side Content
          </div>
        </div>
      </LayoutSidebar>
      <div
        className="sui-layout-main"
      >
        <div
          className="sui-layout-main-header"
        >
          <div
            className="sui-layout-main-header__inner"
          >
            <div>
              Body Header
            </div>
          </div>
        </div>
        <div
          className="sui-layout-main-body"
        >
          <div>
            Body Content
          </div>
        </div>
        <div
          className="sui-layout-main-footer"
        >
          <div>
            Body Footer
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`will accept children instead of bodyContent 1`] = `
<div
  className="sui-layout"
>
  <div
    className="sui-layout-header"
  >
    <div
      className="sui-layout-header__inner"
    >
      <div>
        Header
      </div>
    </div>
  </div>
  <div
    className="sui-layout-body"
  >
    <div
      className="sui-layout-body__inner"
    >
      <LayoutSidebar
        className="sui-layout-sidebar"
      >
        <div>
          <div>
            Side Content
          </div>
        </div>
      </LayoutSidebar>
      <div
        className="sui-layout-main"
      >
        <div
          className="sui-layout-main-header"
        >
          <div
            className="sui-layout-main-header__inner"
          >
            <div>
              Body Header
            </div>
          </div>
        </div>
        <div
          className="sui-layout-main-body"
        >
          <div>
            Body Content
          </div>
        </div>
        <div
          className="sui-layout-main-footer"
        >
          <div>
            Body Footer
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
